src/praisonai-agents/praisonaiagents/workspace/__init__.py

```
@@ -0,0 +1,186 @@
+        # If path is absolute, check if it's within workspace
+        if path.is_absolute():
+            resolved = path.resolve()
+        else:
+            # Resolve relative to workspace root
+            resolved = (self.root / path).resolve()
+        
+        # Security check: ensure resolved path is within workspace
+        try:
+            resolved.relative_to(self.root)
+        except ValueError:
+            raise ValueError(f"Path escapes workspace: {path} -> {resolved} (workspace: {self.root})")
```

Ruff (0.15.10)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)